apps/api/R/submit.workflow.R

```diff
@@ -48,23 +48,35 @@ submit.workflow.list <- function(workflowList, userDetails, dbcon = global_db_po
     )
   )
 
-  if (!is.null(workflowList$model$id) &&
-        (is.null(workflowList$model$type) || is.null(workflowList$model$revision))) {
-    res <- dplyr::tbl(dbcon, "models") %>%
-      select(id, model_name, revision) %>%
-      filter(id == !!workflowList$model$id) %>%
-      collect()
-
-    workflowList$model$type <- res$model_name
-    workflowList$model$revision <- res$revision
+  if (is.null(workflowList$model$id)) {
+    stop("Must provide model ID.")
   }
 
+  # Get model revision and type for the RabbitMQ queue
+  model_info <- dplyr::tbl(dbcon, "models") %>%
+    dplyr::filter(id == !!workflowList$model$id) %>%
+    dplyr::inner_join(dplyr::tbl(dbcon, "modeltypes"),
+                      by = c("modeltype_id" = "id")) %>%
+    dplyr::collect()
+
+  if (nrow(model_info) < 1) {
+    stop("No models found with ID ", format(workflowList$model$id, scientific = FALSE))
+  } else if (nrow(model_info) > 1) {
+    stop("Found multiple (", nrow(model_info), ") matching models for id ",
+         format(workflowList$model$id, scientific = FALSE),
+         ". This shouldn't happen! ",
+         "Check your database for errors.")
+  }
```

Comment:
The second error might be overkill, but I feel like it can't hurt to protect against weird errors.